eth/protocols/eth/peer.go

```diff
@@ -348,10 +350,15 @@ func (p *Peer) SendNewBlock(block *types.Block, td *big.Int) error {
 		p.knownBlocks.Pop()
 	}
 	p.knownBlocks.Add(block.Hash())
-	return p2p.Send(p.rw, NewBlockMsg, &NewBlockPacket{
+
+	start := time.Now()
+	err := p2p.Send(p.rw, NewBlockMsg, &NewBlockPacket{
 		Block: block,
 		TD:    td,
 	})
+	perf.RecordMPMetrics(perf.MpPropagationSend, start)
+
+	return err
```

Comment:
It is a bit tricky to get the propagation time, since we need record the sending time in one peer, and record another receiving time in other peers.

Reply:
added logs and fetchHeader/fetchBodies, not sure whether it is correct or not